app/controllers/results_controller.rb

```diff
@@ -6,9 +6,18 @@ def create
     redirect_to result_path(assessment_code:)
   end
 
+  def early_result
+    # hard coding this at the moment - we could look for last step with valid data but that might cause issues if
+    # banner displayed on multiple pages OR we can send the step down in the param?
+    session_data["api_response"] = CfeService.call(session_data, Steps::Helper.completed_steps_for(session_data, :other_income))
+    redirect_to result_path(assessment_code:, early_assessment: :gross_income)
```

Comment:
I was thinking I would rather pass something down the params but could also add to the session - it just means we need to deal with the clean up if a user goes back.

Reply:
changed functionality